Importing Pandas Library.

In [48]:
import pandas as pd

Reading the csv data.

In [2]:
df = pd.read_csv("2022_02_08-02_30_31_AM.csv")

In [3]:
df

uuid price_string price_string_unf  \
0         638744a4-b0ae-4166-8cb6-5c063c862036          NaN              NaN   
1         ab313969-02cc-48b2-9daf-0054efb70b92          NaN              NaN   
2         acbd66ff-79f8-467a-91f9-108a45af5626       $19.95              NaN   
3         963915d6-b2e3-4098-b242-9410593cf205       $92.00              NaN   
4         b5b68f3c-b1e0-40e5-8ee5-e2f7236c1daf        11.50              NaN   
...                                        ...          ...              ...   
13729995  2a13338e-9268-4f7b-9e7c-b8c41c72971b          NaN              NaN   
13729996  fcaedae7-dc20-4da1-b4ae-77368c973088       $69.99              NaN   
13729997  679f073a-862a-4e43-b5ce-41e45afd3bfa          NaN              NaN   
13729998  25677eee-c26b-4abf-bd10-ca77a800ba26          NaN              NaN   
13729999  0f4550de-1f33-48b5-a5ef-fc5e39df208e       $55.00              NaN   

                                               product_type  \
0                                  TGFwdG9wIENvdmVycy9CYWdz   
1                               QmFraW5nIEN1cHMgYW5kIE1hdHM   
2         R3VtbWllcyB2aXRhbWlucyBhbmQgbWluZXJhbHMgZm9yIG...   
3                                                  U2VydW1z   
4                           RWF0aW5nIFV0ZW5zaWxzL0N1dGxlcnk   
...                                                     ...   
13729995                                         RHJlc3Nlcw   
13729996                                     Um9vbSBkZWNvcg   
13729997                                        UGV0IEZvb2Q   
13729998                                            U2F1Y2U   
13729999                             TG9hZmVycy8gT3hmb3Jkcw   

                                      level_1  \
0                             TGFwdG9wIENhc2U   
1          QmFraW5nIE1hdHMgLyBCYWtpbmcgZGlzaA   
2                      SW1tdW5pdHkgZ3VtbWllcw   
3                              RmFjZSBTZXJ1bQ   
4                              Q2hvcHN0aWNrcw   
...                                       ...   
13729995                       TWF4aSBkcmVzcw   
13729996                  UGxhbnRlcnMvVmFzZXM   
13729997                     RHJ5IERvZyBGb29k   
13729998                         QkJRIFNhdWNl   
13729999  d29tZW5zIE94Zm9yZHMgYW5kIExvYWZlcnM   

                                  category  
0                          RWxlY3Ryb25pY3M  
1               a2l0Y2hpbmcgYW5kIGRpbmluZw  
2                                 SGVhbHRo  
3         YmVhdXR5IGFuZCBwZXJzb25hbCBjYXJl  
4               a2l0Y2hpbmcgYW5kIGRpbmluZw  
...                                    ...  
13729995  Q2xvdGhpbmcgYW5kIEFjY2Vzc29yaWVz  
13729996    SG91c2Vob2xkIGFuZCBDbGVhbmluZw  
13729997                  cGV0IHN1cHBsaWVz  
13729998                      Z3JvY2VyaWVz  
13729999  Q2xvdGhpbmcgYW5kIEFjY2Vzc29yaWVz  

[13730000 rows x 6 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13730000 entries, 0 to 13729999
Data columns (total 6 columns):
 #   Column            Dtype 
---  ------            ----- 
 0   uuid              object
 1   price_string      object
 2   price_string_unf  object
 3   product_type      object
 4   level_1           object
 5   category          object
dtypes: object(6)
memory usage: 628.5+ MB


Filling NaN values = 0

In [5]:
df1 = df.fillna(0)

In [6]:
df1

uuid price_string price_string_unf  \
0         638744a4-b0ae-4166-8cb6-5c063c862036            0                0   
1         ab313969-02cc-48b2-9daf-0054efb70b92            0                0   
2         acbd66ff-79f8-467a-91f9-108a45af5626       $19.95                0   
3         963915d6-b2e3-4098-b242-9410593cf205       $92.00                0   
4         b5b68f3c-b1e0-40e5-8ee5-e2f7236c1daf        11.50                0   
...                                        ...          ...              ...   
13729995  2a13338e-9268-4f7b-9e7c-b8c41c72971b            0                0   
13729996  fcaedae7-dc20-4da1-b4ae-77368c973088       $69.99                0   
13729997  679f073a-862a-4e43-b5ce-41e45afd3bfa            0                0   
13729998  25677eee-c26b-4abf-bd10-ca77a800ba26            0                0   
13729999  0f4550de-1f33-48b5-a5ef-fc5e39df208e       $55.00                0   

                                               product_type  \
0                                  TGFwdG9wIENvdmVycy9CYWdz   
1                               QmFraW5nIEN1cHMgYW5kIE1hdHM   
2         R3VtbWllcyB2aXRhbWlucyBhbmQgbWluZXJhbHMgZm9yIG...   
3                                                  U2VydW1z   
4                           RWF0aW5nIFV0ZW5zaWxzL0N1dGxlcnk   
...                                                     ...   
13729995                                         RHJlc3Nlcw   
13729996                                     Um9vbSBkZWNvcg   
13729997                                        UGV0IEZvb2Q   
13729998                                            U2F1Y2U   
13729999                             TG9hZmVycy8gT3hmb3Jkcw   

                                      level_1  \
0                             TGFwdG9wIENhc2U   
1          QmFraW5nIE1hdHMgLyBCYWtpbmcgZGlzaA   
2                      SW1tdW5pdHkgZ3VtbWllcw   
3                              RmFjZSBTZXJ1bQ   
4                              Q2hvcHN0aWNrcw   
...                                       ...   
13729995                       TWF4aSBkcmVzcw   
13729996                  UGxhbnRlcnMvVmFzZXM   
13729997                     RHJ5IERvZyBGb29k   
13729998                         QkJRIFNhdWNl   
13729999  d29tZW5zIE94Zm9yZHMgYW5kIExvYWZlcnM   

                                  category  
0                          RWxlY3Ryb25pY3M  
1               a2l0Y2hpbmcgYW5kIGRpbmluZw  
2                                 SGVhbHRo  
3         YmVhdXR5IGFuZCBwZXJzb25hbCBjYXJl  
4               a2l0Y2hpbmcgYW5kIGRpbmluZw  
...                                    ...  
13729995  Q2xvdGhpbmcgYW5kIEFjY2Vzc29yaWVz  
13729996    SG91c2Vob2xkIGFuZCBDbGVhbmluZw  
13729997                  cGV0IHN1cHBsaWVz  
13729998                      Z3JvY2VyaWVz  
13729999  Q2xvdGhpbmcgYW5kIEFjY2Vzc29yaWVz  

[13730000 rows x 6 columns]

Here we replace the '$' sign by empty space( Removing everything except numeric values).

In [7]:
df2 = pd.to_numeric(df1.price_string.replace('[^0-9\.-]', '', regex=True))

In [8]:
df2

0            0.00
1            0.00
2           19.95
3           92.00
4           11.50
            ...  
13729995     0.00
13729996    69.99
13729997     0.00
13729998     0.00
13729999    55.00
Name: price_string, Length: 13730000, dtype: float64

In [9]:
df1

uuid price_string price_string_unf  \
0         638744a4-b0ae-4166-8cb6-5c063c862036            0                0   
1         ab313969-02cc-48b2-9daf-0054efb70b92            0                0   
2         acbd66ff-79f8-467a-91f9-108a45af5626       $19.95                0   
3         963915d6-b2e3-4098-b242-9410593cf205       $92.00                0   
4         b5b68f3c-b1e0-40e5-8ee5-e2f7236c1daf        11.50                0   
...                                        ...          ...              ...   
13729995  2a13338e-9268-4f7b-9e7c-b8c41c72971b            0                0   
13729996  fcaedae7-dc20-4da1-b4ae-77368c973088       $69.99                0   
13729997  679f073a-862a-4e43-b5ce-41e45afd3bfa            0                0   
13729998  25677eee-c26b-4abf-bd10-ca77a800ba26            0                0   
13729999  0f4550de-1f33-48b5-a5ef-fc5e39df208e       $55.00                0   

                                               product_type  \
0                                  TGFwdG9wIENvdmVycy9CYWdz   
1                               QmFraW5nIEN1cHMgYW5kIE1hdHM   
2         R3VtbWllcyB2aXRhbWlucyBhbmQgbWluZXJhbHMgZm9yIG...   
3                                                  U2VydW1z   
4                           RWF0aW5nIFV0ZW5zaWxzL0N1dGxlcnk   
...                                                     ...   
13729995                                         RHJlc3Nlcw   
13729996                                     Um9vbSBkZWNvcg   
13729997                                        UGV0IEZvb2Q   
13729998                                            U2F1Y2U   
13729999                             TG9hZmVycy8gT3hmb3Jkcw   

                                      level_1  \
0                             TGFwdG9wIENhc2U   
1          QmFraW5nIE1hdHMgLyBCYWtpbmcgZGlzaA   
2                      SW1tdW5pdHkgZ3VtbWllcw   
3                              RmFjZSBTZXJ1bQ   
4                              Q2hvcHN0aWNrcw   
...                                       ...   
13729995                       TWF4aSBkcmVzcw   
13729996                  UGxhbnRlcnMvVmFzZXM   
13729997                     RHJ5IERvZyBGb29k   
13729998                         QkJRIFNhdWNl   
13729999  d29tZW5zIE94Zm9yZHMgYW5kIExvYWZlcnM   

                                  category  
0                          RWxlY3Ryb25pY3M  
1               a2l0Y2hpbmcgYW5kIGRpbmluZw  
2                                 SGVhbHRo  
3         YmVhdXR5IGFuZCBwZXJzb25hbCBjYXJl  
4               a2l0Y2hpbmcgYW5kIGRpbmluZw  
...                                    ...  
13729995  Q2xvdGhpbmcgYW5kIEFjY2Vzc29yaWVz  
13729996    SG91c2Vob2xkIGFuZCBDbGVhbmluZw  
13729997                  cGV0IHN1cHBsaWVz  
13729998                      Z3JvY2VyaWVz  
13729999  Q2xvdGhpbmcgYW5kIEFjY2Vzc29yaWVz  

[13730000 rows x 6 columns]

In [10]:
average = df2.mean()

In [11]:
print("Average price of products is:\n", average)

Average price of products is:
 28.103401310997775


In [12]:
products_arr = df1["uuid"].unique()

In [13]:
print(products_arr)

['638744a4-b0ae-4166-8cb6-5c063c862036'
 'ab313969-02cc-48b2-9daf-0054efb70b92'
 'acbd66ff-79f8-467a-91f9-108a45af5626' ...
 '62785956-705e-47db-bd0e-764575333724'
 'b4799645-56ab-4b83-8666-1b79cbf6da17'
 '21e8ead0-6f8a-40aa-9ad9-469abd8dad3d']


In [51]:
print("Number of unique products is:\n", len(products_arr))

Number of unique products is:
 1373


In [15]:
categories_arr = df1["category"].unique()

In [16]:
print(categories_arr)

['RWxlY3Ryb25pY3M' 'a2l0Y2hpbmcgYW5kIGRpbmluZw' 'SGVhbHRo'
 'YmVhdXR5IGFuZCBwZXJzb25hbCBjYXJl' 'Z3JvY2VyaWVz'
 'SG91c2Vob2xkIGFuZCBDbGVhbmluZw' 'Q2xvdGhpbmcgYW5kIEFjY2Vzc29yaWVz'
 'VG95cyBhbmQgR2FtZXM' 'QXJ0IHN1cHBsaWVz' 'cGV0IHN1cHBsaWVz'
 'VG9vbHMgYW5kIGhvbWUgaW1wcm92ZW1lbnQ' 'QmFieWNhcmU'
 'b2ZmaWNlIHByb2R1Y3Rz' 0]


In [52]:
print("Number of unique categories is:\n",len(categories_arr))

Number of unique categories is:
 14


In [18]:
product_type_arr = df1["product_type"].unique()

In [19]:
print(product_type_arr)

['TGFwdG9wIENvdmVycy9CYWdz' 'QmFraW5nIEN1cHMgYW5kIE1hdHM'
 'R3VtbWllcyB2aXRhbWlucyBhbmQgbWluZXJhbHMgZm9yIGtpZHM' 'U2VydW1z'
 'RWF0aW5nIFV0ZW5zaWxzL0N1dGxlcnk' 'TmF0dXJhbCBTd2VldGVuZXJzLyBTdWdhcg'
 'TW9wcyBhbmQgYnJvb21z' 'V29tZW5zIFBhbnRz' 'Um9sbGluZyBQaW4'
 'Q2FyZHN0b2Nrcw' 'Rm9vZCBTdG9yYWdl' 'U2NydWJzIGFuZCBjbGVhbmluZyBjbG90aHM'
 'U2hvcnRz' 'UGxhdGVzL1V0ZW5zaWxzL0N1dGxlcnk'
 'UGx1cyBzaXplIHdlYXIgZHJlc3M' 'TnV0cyAtIENhc2hld3M'
 'TWFrZXVwIGFwcGxpY2F0b3I' 'TWF0ZXJuaXR5IFBhZHM'
 'RWR1Y2F0aW9uYWwgQmxvY2tzL1RveXM'
 'UGxhbnQgYmFzZWQgUHJvdGVpbiBTdXBwbGVtZW50cw' 'Q2FuZGxlcy9GcmFnbmFuY2Vz'
 'QmVkc2hlZXRz' 'bWVucyBCb2R5IFNvYXAvIEJvZHl3YXNo' 'TGVnb3M'
 'U3VyZiBzdWl0cw' 'UmljZQ' 'RG9sbCBob3VzZSAvIEZpZ3VyZXM' 'UGV0IGNvbGxhcnM'
 'TmF0dXJhbCBQaWdtZW50cyA' 'UG9wY29ybg' 'U29sYXItUG93ZXJlZCBGbGFzaGxpZ2h0'
 'U2F1Y2U' 'VGhlcm1vc3RhdHM' 'QnJ1c2hlcw'
 'UHJvZHVjZSBCYWdzIGFuZCBHcm9jZXJ5IFRvdGUgQmFncw' 'U2xlZXAgc2hvcnRz'
 'RXllIG1hc2s' 'V29tZW5zIHN3aW13ZWFy' 'V29tZW5zIEphY2tldHM'
 'U3F1ZWdlZSBhbmQgZHVzdGVy' 'U

In [54]:
print("Number of unique product type is:\n",len(product_type_arr))

Number of unique product type is:
 322


In [24]:
level_1_arr = df1["level_1"].unique()

In [25]:
print(level_1_arr)

['TGFwdG9wIENhc2U' 'QmFraW5nIE1hdHMgLyBCYWtpbmcgZGlzaA'
 'SW1tdW5pdHkgZ3VtbWllcw' 'RmFjZSBTZXJ1bQ' 'Q2hvcHN0aWNrcw'
 'TW9uayBGcnVpdA' 'TW9w' 'V29tZW5zIFJlZ3VsYXIvQ2FzdWFsIFBhbnRz'
 'V29tZW5zIFRpZ2h0cw' 'UGxheSBEb3VnaCBSb2xsaW5nIFBpbg' 'Q2FyZHM'
 'Rm9vZCBDb250YWluZXI' 'U2NydWJz' 'V29tZW5zIFNob3J0cw' 'UGxhdGVz'
 'RHJlc3M' 'Q2FzaGV3cw' 'TWFrZXVwIGFwcGxpY2F0b3I' 'UG9zdHBhcnR1bSBQYWRz'
 'TnVtYmVyIFRveXM'
 'S2V0byBEaWV0IFdlaWdodCBsb3NzIHBsYW50IGJhc2VkIHByb3RlaW4gcG93ZGVy'
 'Q2FuZGxlcw' 'UGVyY2FsZSBiZWRzaGVldHM' 'bWVucyBCb2R5d2FzaA'
 'TGVnb3MgYW5pbWFscw' 'V29tZW5zIHN1cmYgc3VpdA' 'V2lsZCBSaWNl'
 'QWN0aW9uIEZpZ3VyZXM' 'Q2F0IGNvbGxhcnM' 'V2F0ZXJjb2xvcnMgcGFpbnRz'
 'UG9wcGVkIFBvcGNvcm4' 0 'QkJRIFNhdWNl' 'VGhlcm1vc3RhdHM' 'RGlzaCBicnVzaA'
 'R3JvY2VyeS9TaG9wcGluZyBUb3RlIEJhZ3M' 'U2xlZXAgc2hvcnRz'
 'Q2FuZGxlIGhvbGRlcg' 'RXllIE1hc2tzL1BhdGNoZXM' 'U3VyZnN1aXQvV2V0c3VpdA'
 'RGVuaW0gSmFja2V0LyBUcnVja2VyIEphY2tldA' 'RHVzdGVy'
 'cGVyY2FsZSBwaWxsb3djYXNl' 'UGFudHkgTGluZXJz' 'R3Jhbm9sYSBCYXJz'
 'V29tZW5zIFN

In [55]:
print("Number of unique level is:\n",len(level_1_arr))

Number of unique level is:
 596


In [28]:
uuid_ser = df1.iloc[:, 0]

In [29]:
uuid_ser

0           638744a4-b0ae-4166-8cb6-5c063c862036
1           ab313969-02cc-48b2-9daf-0054efb70b92
2           acbd66ff-79f8-467a-91f9-108a45af5626
3           963915d6-b2e3-4098-b242-9410593cf205
4           b5b68f3c-b1e0-40e5-8ee5-e2f7236c1daf
                            ...                 
13729995    2a13338e-9268-4f7b-9e7c-b8c41c72971b
13729996    fcaedae7-dc20-4da1-b4ae-77368c973088
13729997    679f073a-862a-4e43-b5ce-41e45afd3bfa
13729998    25677eee-c26b-4abf-bd10-ca77a800ba26
13729999    0f4550de-1f33-48b5-a5ef-fc5e39df208e
Name: uuid, Length: 13730000, dtype: object

In [33]:
uuid_ser.value_counts(sort=True)

638744a4-b0ae-4166-8cb6-5c063c862036    10000
32a6ce63-6ecd-46c5-84d9-4487bb13dd8b    10000
b524d72b-11a5-417b-be83-b33a83c81f4e    10000
2c7e75c2-c239-4461-a231-810c1dbcb51e    10000
08b5646e-16dd-4a6e-b0d8-0394a0f10d82    10000
                                        ...  
09f79df9-4500-48f6-9941-8c05c7c241ae    10000
6202649f-4bf3-41f0-ab94-ae26f68e7e89    10000
61e4f1c3-7c2d-45cf-b9e4-2ef9ad82c96f    10000
06b44dd1-25f9-463b-9ae4-b0f5df9646bc    10000
21e8ead0-6f8a-40aa-9ad9-469abd8dad3d    10000
Name: uuid, Length: 1373, dtype: int64

THERE ARE 10000 UNIT OF 1373 PRODUCTS hence total 13730000 PRODUCTS.

In [34]:
price_string_ser = df1.iloc[:, 1]

In [35]:
price_string_ser

0                0
1                0
2           $19.95
3           $92.00
4            11.50
             ...  
13729995         0
13729996    $69.99
13729997         0
13729998         0
13729999    $55.00
Name: price_string, Length: 13730000, dtype: object

In [36]:
price_string_ser.value_counts(sort=True)

0          8220000
$0.00       240000
$89         120000
$9.99       100000
$3.99        90000
            ...   
$18          10000
$7.49        10000
$6.25        10000
$299         10000
$268.00      10000
Name: price_string, Length: 249, dtype: int64

WE CAN SEE PRODUCT WITHOUT PRICES ARE 8220000 AND REST 5510000 ARE WITH PRICE.

In [37]:
product_type_ser = df1.iloc[:, 3]

In [38]:
product_type_ser

0                                    TGFwdG9wIENvdmVycy9CYWdz
1                                 QmFraW5nIEN1cHMgYW5kIE1hdHM
2           R3VtbWllcyB2aXRhbWlucyBhbmQgbWluZXJhbHMgZm9yIG...
3                                                    U2VydW1z
4                             RWF0aW5nIFV0ZW5zaWxzL0N1dGxlcnk
                                  ...                        
13729995                                           RHJlc3Nlcw
13729996                                       Um9vbSBkZWNvcg
13729997                                          UGV0IEZvb2Q
13729998                                              U2F1Y2U
13729999                               TG9hZmVycy8gT3hmb3Jkcw
Name: product_type, Length: 13730000, dtype: object

In [39]:
product_type_ser.value_counts(sort=True)

U2NydWJzIGFuZCBjbGVhbmluZyBjbG90aHM           200000
V29tZW5zIHN3aW13ZWFy                          190000
V29tZW5zIHdvcmt3ZWFyLyBvZmZpY2Ugd2Vhcg        170000
UGxhbnQgYmFzZWQgUHJvdGVpbiBTdXBwbGVtZW50cw    150000
d29tZW5zIE91dGVyd2Vhcg                        150000
                                               ...  
SW50aW1hdGUgV2lwZXM                            10000
QmFieSBDcmVhbXMgYW5kIExvdGlvbnM                10000
bWVucyBTaGF2aW5nIEtpdA                         10000
bWVucyBCZWFyZCBDYXJlIEtpdA                     10000
QmFieSBTb2Fw                                   10000
Name: product_type, Length: 322, dtype: int64

PRODUCT TYPE WISE PRODUCT COUNT

In [58]:
level_1_ser = df1.iloc[:, 4]

In [41]:
level_1_ser

0                               TGFwdG9wIENhc2U
1            QmFraW5nIE1hdHMgLyBCYWtpbmcgZGlzaA
2                        SW1tdW5pdHkgZ3VtbWllcw
3                                RmFjZSBTZXJ1bQ
4                                Q2hvcHN0aWNrcw
                           ...                 
13729995                         TWF4aSBkcmVzcw
13729996                    UGxhbnRlcnMvVmFzZXM
13729997                       RHJ5IERvZyBGb29k
13729998                           QkJRIFNhdWNl
13729999    d29tZW5zIE94Zm9yZHMgYW5kIExvYWZlcnM
Name: level_1, Length: 13730000, dtype: object

In [42]:
level_1_ser.value_counts(sort=True)

U3dlYXRzaGlydA                  90000
U29ja3M                         70000
Q2xlYW5pbmcgY2xvdGg             70000
U3Bvb24sIEtuaWZlIGFuZCBGb3Jr    70000
TG9uZyBzbGVldmUgdGVl            60000
                                ...  
VHJhc2ggQmFncw                  10000
UGlsbG93L0N1c2hpb24             10000
TG9vbXM                         10000
TWVucyBzbmVha2Vycw              10000
UHVwcHkgU2hhbXBvbw              10000
Name: level_1, Length: 596, dtype: int64

LEVEL_1 WISE PRODUCT COUNT

In [43]:
category_ser = df1.iloc[:, 5]

In [44]:
category_ser

0                            RWxlY3Ryb25pY3M
1                 a2l0Y2hpbmcgYW5kIGRpbmluZw
2                                   SGVhbHRo
3           YmVhdXR5IGFuZCBwZXJzb25hbCBjYXJl
4                 a2l0Y2hpbmcgYW5kIGRpbmluZw
                          ...               
13729995    Q2xvdGhpbmcgYW5kIEFjY2Vzc29yaWVz
13729996      SG91c2Vob2xkIGFuZCBDbGVhbmluZw
13729997                    cGV0IHN1cHBsaWVz
13729998                        Z3JvY2VyaWVz
13729999    Q2xvdGhpbmcgYW5kIEFjY2Vzc29yaWVz
Name: category, Length: 13730000, dtype: object

In [45]:
category_ser.value_counts(sort=True)

Q2xvdGhpbmcgYW5kIEFjY2Vzc29yaWVz       3910000
Z3JvY2VyaWVz                           1610000
SG91c2Vob2xkIGFuZCBDbGVhbmluZw         1570000
YmVhdXR5IGFuZCBwZXJzb25hbCBjYXJl       1470000
SGVhbHRo                                990000
VG95cyBhbmQgR2FtZXM                     790000
cGV0IHN1cHBsaWVz                        740000
QmFieWNhcmU                             660000
a2l0Y2hpbmcgYW5kIGRpbmluZw              590000
VG9vbHMgYW5kIGhvbWUgaW1wcm92ZW1lbnQ     520000
RWxlY3Ryb25pY3M                         320000
QXJ0IHN1cHBsaWVz                        320000
b2ZmaWNlIHByb2R1Y3Rz                    230000
0                                        10000
Name: category, dtype: int64

CATEGORY WISE PRODUCT COUNT